source/common/upstream/upstream_impl.cc

```diff
@@ -262,12 +262,34 @@ ResourceManager& ClusterInfoImpl::resourceManager(ResourcePriority priority) con
   return *resource_managers_.managers_[enumToInt(priority)];
 }
 
-void ClusterImplBase::blockHcUpdates(bool block) {
-  ASSERT(block_hc_updates_ == !block);
-  block_hc_updates_ = block;
+void ClusterImplBase::setInitialized() {
```

Comment:
I think we can make this function a bit easier to understand.    As I read it, if we're doing health checking we expect and assert setInitialized() to be called exactly twice, otherwise it'll be called once.    The first time by the cluster impl, and the maybe-second when the last health check is complete?

both times we'll set initialized_ to true (so we consider the cluster initialized even without outstanding health checks) but we only call initialized_ health checks after all the health checks are done (despite being "initialized" the first time it's called)

I think it'd be a bit more clear if we (optionally) split it into startInitialization() and finishInitialization() (where you call start once, and finish once)
renamed initialized_ -> initialization_started_ and
initialization_callback_ -> initialization_compelte_callback_
as I think that's what maps logically to what's going on.  WTYT?

Reply:
I was going for limited code change but I agree that would be clearer. Will do.